workorder-detail-fix.lua

```diff
@@ -0,0 +1,114 @@
+local script_name = "workorder-detail-fix"
+local eventful = require 'plugins.eventful'
+if not handler_ref then local handler_ref = nil end 
+
+local function get_job_id(match)
+    for val, name in ipairs(df.job_type) do 
+        if name == match then return val end
+    end
+end
+
+-- all jobs with the "any" (-1) type in its default job_items may be a problem
+local offending_jobs = {
+    [get_job_id("EncrustWithGems")] = true,

```

Comment:
you can just do `df.job_type.EncrustWithGems` etc instead.

btw,
```lua
type( df.job_type.EncrustWithGems ) -- 86, number
type( df.job_type[86] ) -- EncrustWithGems, string
```

Reply:
Much better, it's done. But i accidentally changed offending_jobs to a non-local variable in the commit. Does that matter? I've seen other scripts do it, but I haven't noticed a pattern as to why

Reply:
generally speaking, it's *usually* better to use local variables where possible, as it usually leads to cleaner code (and also helps avoid bugs if various parts of a program use the same global variable).

Also, in Lua accessing a local variable is faster. You'd see stuff like `local print = print` before tight loops for this reason.

As for reasons to use globals in dfhack scripts specifically: the value is then stored between script executions. This can be useful f.e. for the script's configuration.

In your specific case, it doesn't matter too much (but ultimately it'd probably be @myk002 who'll approve this PR, so it's actually up to them). Personally, I would use a local (because the script as a whole won't run often, only the one function from it). Also personally, in case I do decide to make it global, I'd make it like this:

```lua
offending_jobs = offending_jobs or {
   [df.job_type.EncrustWithGems] = true,
   -- ... etc
}
```

This reuses a previously created table `offending_jobs` if there already is one, or creates one if this is the first time the script is run (and the value of `offending_jobs` is `nil` when we read on the right side of that assignment).